pkg/controllers/repositoryserver/suite_test.go

```diff
@@ -15,58 +15,174 @@
-	if s.testEnv != nil {
-		c.Log("Tearing down the test environment")
-		err := s.testEnv.Stop()
-		c.Assert(err, IsNil)
+func (s *RepoServerControllerSuite) createRepositoryServerSecrets(c *C) {
+	repoServerUserAccessSecretData := map[string][]byte{
+		"localhost": []byte(DefaultKopiaRepositoryServerAccessPassword),
 	}
+	repoServerAdminSecretData := map[string][]byte{
+		"username": []byte(DefaulKopiaRepositoryServerAdminUser),
+		"password": []byte(DefaultKopiaRepositoryServerAdminPassword),
+	}
+	repoPasswordSecretData := map[string][]byte{
+		repoPasswordKey: []byte(DefaultKopiaRepositoryPassword),
+	}
+	kopiaTLSSecretData, err := getKopiaTLSSecret()
+	c.Assert(err, IsNil)
+	s.repoServerSecrets = repositoryServerSecrets{}
+	s.repoServerSecrets.serverUserAccess, err = s.createSecret("test-repository-server-user-access-", "", repoServerUserAccessSecretData)
+	c.Assert(err, IsNil)
+	s.repoServerSecrets.serverAdmin, err = s.createSecret("test-repository-server-admin-", "", repoServerAdminSecretData)
+	c.Assert(err, IsNil)
+	s.repoServerSecrets.repositoryPassword, err = s.createSecret("test-repository-password-", "", repoPasswordSecretData)
+	c.Assert(err, IsNil)
+	s.repoServerSecrets.serverTLS, err = s.createSecret("test-tls-", v1.SecretTypeTLS, kopiaTLSSecretData)
+	c.Assert(err, IsNil)
+	s.repoServerSecrets.storage, err = s.createSecret("test-repository-server-storage-", "", getDefaultS3StorageLocation())
+	c.Assert(err, IsNil)
+	s.repoServerSecrets.storageCredentials, err = s.createSecret("test-repository-server-storage-Creds-", "secrets.kanister.io/aws", getDefaultS3StorageCreds())
+	c.Assert(err, IsNil)
 }
 
-func (s *ControllerSuite) TestWatch(c *C) {
-	// We give it a few seconds complete it's scan. This isn't required for the
-	// test, but is a more realistic startup scenario.
-	time.Sleep(5 * time.Second)
+func (s *RepoServerControllerSuite) createSecret(name string, secrettype v1.SecretType, data map[string][]byte) (se *v1.Secret, err error) {
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: name,
+		},
+		Data: data,
+	}
+	if secrettype != "" {
+		secret.Type = secrettype
+	}
+
+	se, err = s.cli.CoreV1().Secrets(s.repoServerControllerNamespace).Create(context.Background(), secret, metav1.CreateOptions{})
+	return
+}
+
+func (s *RepoServerControllerSuite) TestRepositoryServerReady(c *C) {
+	repoServerCR := getDefaultKopiaRepositoryServerCR(s.repoServerControllerNamespace)
+	setRepositoryServerSecretsInCR(&s.repoServerSecrets, repoServerCR)
+	repoServerCRCreated, err := s.crCli.RepositoryServers(s.repoServerControllerNamespace).Create(context.Background(), repoServerCR, metav1.CreateOptions{})
+	c.Assert(err, IsNil)
+	err = s.waitForRepoServerInfoUpdateInCR(repoServerCRCreated)
+	c.Assert(err, IsNil)
+	err = createKopiaRepository(s.cli, repoServerCRCreated, getDefaultS3StorageLocation())
+	c.Assert(err, IsNil)
+	err = s.waitOnRepositoryServerState(c, repoServerCRCreated)
+	c.Assert(err, IsNil)
```

Comment:
The function says we are waiting for RepoServer to be ready but here we are waiting for the server to get created? Seems confusing.

Reply:
I have renamed the test